test/unit/api/factory/ItemCategory.test.ts

```diff
@@ -0,0 +1,72 @@
+import { LogMock } from '../../lib/LogMock';
+import { ItemCategoryFactory } from '../../../../src/api/factories/ItemCategoryFactory';
+
+describe('ItemCategoryFactory', () => {
+
+    let req;
+    let itemCategoryFactory;
+    const rootCategoryWithRelated = {
+        id: 1,
+        Key: 'cat_ROOT',
+        name: 'ROOT',
+        ChildItemCategories: [
+            {
+                id: 2,
+                Key: 'cat_high_value',
+                name: 'High Value',
+                parentItemCategoryId: 1,
+                ChildItemCategories: [
+                    {
+                        id: 3,
+                        Key: 'cat_high_business_corporate',
+                        name: 'Business Corporate',
+                        parentItemCategoryId: 2,
+                        ChildItemCategories: []
+                    }
+                ]
+            }
+        ]
+    };
+    beforeEach(() => {
+        process.env.AUTH0_HOST = 'test';
+        itemCategoryFactory = new ItemCategoryFactory(LogMock);
+    });
+
+    test('Should get the item-category data when pass root category only', () => {
+        req = ['cat_ROOT', 'Subcategory', 'SubSubcategory'];
+        itemCategoryFactory.get(req, rootCategoryWithRelated).then((res, error) => {
+            expect(res.length).toBe(2);
+            expect(res[0].parent_item_category_id).not.toBe(0);
+            expect(res[0].name).toBe(req[1]);
+
+            expect(res[1].parent_item_category_id).toBe('0');
+            expect(res[1].name).toBe(req[2]);
```

Comment:
why is parent_item_category_id expected to be a string and '0'
id's are always numbers and cat_ROOT should have parent_item_category_id === null